sirepo/package_data/static/js/raydata.js

```diff
@@ -488,6 +502,17 @@ SIREPO.app.directive('scansTable', function() {
                 $scope.saveColumnChanges();
             };
 
+            $scope.disableRunAnalysis = (scan) => {
```

Comment:
I think this can be reduced to just `return ["pending", "running"].includes(scan.status)`.

Then in runAnalysis instead of adding the scan to disabledRunButtonScans we can set the scan.status to pending. 

That way we eliminate the logic for keeping track of disabledRunButtonScans.

Reply:
This doesn't work - the front end receives another response that resets the scan status before the server has changed it to pending, which re-enables the button for a moment. I think the front end needs to keep track of this.

Reply:
Ah right. Sorry about that.

One concern with this code is if a scan doesn't enter an `ANALYSIS_STATUS_NON_STOPPED` state in time for the GUI to pick up on it it will never be removed from disabledRunButtonScans. For example, this could happen if it goes straight to error w/o a call to run_status in between to pick up on the temporary running/pending state.

Here is a strategy I think might work: On sending to run_analysis add the scan to a dict of `disabledRunButtonScans` (maybe rename to pendingRunAnalysis or something like that). On reply from the server (error or success) remove the scan from the dict, set its status to pending. On the next scans request the status will be updated with the actual status of the scan (what the scan monitor says). In $scope.disableRunAnalysis() check pendingRunAnalysis. If it is in pendingRunAnalysis then return true. Else return raydataService.ANALYSIS_STATUS_NON_STOPPED.includes(scan.status).


Also, a dict should be used instead of an array to keep track of the disable scans. The access pattern is to always look for a certain scan id. So, we can have that be the key and the value doesn't matter (ex true for everything). That way we do constant time lookups/deletions instead of linear time (includes, splice).

Reply:
260c46d618791faf0cbf7814cf1b64c872b72c51